Source/Core/ClippingPlaneCollection.js

```diff
@@ -234,53 +346,203 @@ define([
      * @see ClippingPlaneCollection#remove
      */
     ClippingPlaneCollection.prototype.removeAll = function() {
+        // Dereference this ClippingPlaneCollection from all ClippingPlanes
+        var planes = this._planes;
+        var planesCount = planes.length;
+        for (var i = 0; i < planesCount; ++i) {
+            var plane = planes[i];
+            if (plane instanceof ClippingPlane) {
+                plane.onChangeCallback = undefined;
+                plane.index = -1;
+            }
+        }
+        this._manyDirtyPlanes = true;
         this._planes = [];
     };
 
-    var scratchPlane = new Plane(Cartesian3.UNIT_X, 0.0);
-    var scratchMatrix = new Matrix4();
+    var octEncodeScratch = new Cartesian2();
+    var rightShift = 1.0 / 256.0;
     /**
-     * Applies the transformations to each plane and packs it into an array.
-     * @private
-     *
-     * @param {Matrix4} viewMatrix The 4x4 matrix to transform the plane into eyespace.
-     * @param {Cartesian4[]} [array] The array into which the planes will be packed.
-     * @returns {Cartesian4[]} The array of packed planes.
+     * Encodes a normalized vector into 4 SNORM values in the range [0-255] following the 'oct' encoding.
+     * oct32 precision is higher than the default oct16, hence the additional 2 uint16 values.
      */
-    ClippingPlaneCollection.prototype.transformAndPackPlanes = function(viewMatrix, array) {
-        //>>includeStart('debug', pragmas.debug);
-        Check.typeOf.object('viewMatrix', viewMatrix);
-        //>>includeEnd('debug');
+    function oct32EncodeNormal(vector, result) {
```

Comment:
Is this specific to the clipping plane?  Or should this be a utility function?  Maybe a static function in `AttributeCompression`?

Reply:
That makes a lot of sense, but it seems like we'd also want to rename a bunch of stuff in `AttributeCompression` to indicate 16-bit-vec2, which seems like it could be a pretty big breaking change.

@lilleyse what do you think? I guess we could also deprecate the current functions over time?